docker/all-in-one/shutdown_event_listener.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python3
+
+# This script is used to make sure the latest LSN checkpoint is persisted remotely
+#  before the container is stopped. It is triggered every 60 seconds, whichever comes first.
+# The script will ship the latest LSN checkpoint to the remote storage if:
+# - the latest LSN checkpoint is different from the previous, already shipped, one
+# - the latest LSN checkpoint is not 0/0 as to avoid shipping empty checkpoints
+# - the latest LSN checkpoint is not older than 10 minutes, thus reducing remote write calls
+
+import os
+import sys
+import subprocess
+import time
+
+LSN_CHECKPOINT_SHIP_INTERVAL = 10
+
+checkpointFile = '/data/latest-lsn-checkpoint'
+checkpointFilePrevious = '/data/latest-lsn-checkpoint.previous'
+
+def write_stdout(s):
+    sys.stdout.write(s)
+    sys.stdout.flush()
+
+def write_stderr(s):
+    sys.stderr.write(s)
+    sys.stderr.flush()
+
+def main():
+    while 1:
+        write_stdout('READY\n')
```

Comment:
This output is a contract with supervisord?

Reply:
do we need additional debugging output to differentiate when the script didn't persist things for w/e reason?

Reply:
>This output is a contract with supervisord?

Yep - `READY` means supervisord can send events
http://supervisord.org/events.html#event-listener-states

End-state of an event completing succesfully is 
```RESULT <resultLen>
<result>```

>do we need additional debugging output to differentiate when the script didn't persist things for w/e reason?
Yep. Would definitely be worth having it log to its own log file